build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/RestTestUtil.java

```diff
@@ -48,7 +48,7 @@ private RestTestUtil() {}
      * Setup the dependencies needed for the REST tests.
      */
     public static void setupTestDependenciesDefaults(Project project, SourceSet sourceSet) {
-        project.getDependencies().add(sourceSet.getImplementationConfigurationName(), project.project(":test:framework"));
+        project.getDependencies().add(sourceSet.getImplementationConfigurationName(), project.project(":test:rest-runner"));
```

Comment:
I've split this out so that the java rest tests have a separate method which still just includes the test framework, and yaml rest tests include the yaml-rest-runner.

Reply:
This ended up generating a slew of new compile failures. I'm going to leave this as java rest tests including the yaml rest runner for now. I'll followup with a separate change to reduce it back to framework.